version.go

```diff
@@ -162,13 +174,33 @@ func semanticVersion() string {
 	// Start with the major, minor, and patch versions.
 	version := fmt.Sprintf("%d.%d.%d", AppMajor, AppMinor, AppPatch)
 
-	// Append pre-release version if there is one. The hyphen called for
-	// by the semantic versioning spec is automatically appended and should
-	// not be contained in the pre-release string. The pre-release version
+	// If defined, we will now sanitise the release status string. The
+	// hyphen called for by the semantic versioning spec is automatically
+	// appended and should not be contained in the status string. The status
 	// is not appended if it contains invalid characters.
+	appStatus := normalizeVerString(AppStatus, semanticAlphabet)
+
+	// If defined, we will now sanitise the pre-release version string. The
+	// hyphen called for by the semantic versioning spec is automatically
+	// appended and should not be contained in the pre-release string.
+	// The pre-release version is not appended if it contains invalid
+	// characters.
 	preRelease := normalizeVerString(AppPreRelease, semanticAlphabet)
-	if preRelease != "" {
-		version = fmt.Sprintf("%s-%s", version, preRelease)
+
+	// Append any status and pre-release strings to the version string.
+	switch {
```

Comment:
Bit confused on this switch, do we ever want appStatus without preRelease? From the tapd tags it looks like we only do both or neither, ex. `v0.3.2`, `v0.3.0-alpha.rc3`, but no `v0.3.0-alpha` nor `v0.3.0-rc3`.

Or is this to also match a different pattern used for a different project?

Reply:
> Or is this to also match a different pattern used for a different project?

I think we should make sure that possibility is available to us. I don't think we have a strict org wide rule around the use of status without pre-release. `v0.3.0-alpha` seems like a reasonable tag to me for instance.

Reply:
K, I thought leaving out the `rc1` suffix when using `alpha` or `beta` is what caused issues previously? I don't recall exactly which tag caused those though.

Reply:
What caused the issue previously was leaving out `-alpha` while just using an `rc` tag (e.g. `0.3.2-rc2`), because then the version check of the release script didn't work as it just did a substring comparison with what the app version string reported (`v0.3.2-alpha.rc2`).
So I think we should also update the `check_tag_correct` function in `scripts/release.sh` to use the new `get-git-tag-name.sh` script and compare it to the checked out tag.

And to answer your other question, just having the status is what we want for non-RC releases in all other projects than this one (e.g. lnd).